right/src/slave_scheduler.c

```diff
@@ -114,6 +116,9 @@ void InitSlaveScheduler(void)
 
     for (uint8_t i=0; i<SLAVE_COUNT; i++) {
         uhk_slave_t *currentSlave = Slaves + i;
+        if (currentSlave->isConnected && currentSlave->disconnect) {
```

Comment:
Shouldn't all slaves be disconnected upon scheduler initialization?

Reply:
Sorry, I guess https://github.com/UltimateHackingKeyboard/firmware/issues/399#issuecomment-1173870244 is the explanation.

Reply:
> Sorry, I guess https://github.com/UltimateHackingKeyboard/firmware/issues/399#issuecomment-1173870244 is the explanation.

Is it?

> Shouldn't all slaves be disconnected upon scheduler initialization?

Should they? Those which are not "connected" are guaranteed to have already been properly disconnected, or never even were connected.

Reply:
Oh, sorry, I see the misunderstanding now. Yes, we are talking re-initializations.